### DISCUSSION TITLE
How can i use reth crate to build custom node ?

### DISCUSSION THREAD

**Discussion by Blaeaea** [paradigmxyz/reth#5049] [Help] [answered]
Hello i wanted to know how can i use reth crates to build custom node, node that can connect to p2p and collect mempool data for example. Can someone help me with that?

**Comment by mattsse** [comment]
looks like you're only interested in the network component:

https://github.com/paradigmxyz/reth/blob/main/examples/network-txpool.rs

**Comment by jmcph4** [comment, marked as answer]
Alternatively, there are some relevant examples, namely:

 - [`custom-node`](https://github.com/paradigmxyz/reth/tree/83cd848639922535dd32c99328411a0bb2440a65/examples/custom-node)
 - [`custom-node-components`](https://github.com/paradigmxyz/reth/tree/83cd848639922535dd32c99328411a0bb2440a65/examples/custom-node-components)
 - [`custom-payload-builder`](https://github.com/paradigmxyz/reth/tree/83cd848639922535dd32c99328411a0bb2440a65/examples/custom-payload-builder)
 - [`custom-dev-node`](https://github.com/paradigmxyz/reth/tree/83cd848639922535dd32c99328411a0bb2440a65/examples/custom-dev-node)
 - [`manual-p2p`](https://github.com/paradigmxyz/reth/tree/83cd848639922535dd32c99328411a0bb2440a65/examples/manual-p2p)

There was also a recent documentation update in #7348 that should prove very useful.
